lib/plugins/chatops/slack.rb

```diff
@@ -0,0 +1,60 @@
+require 'slack-notifier'
+
+class PuppetWebhook
+  class Chatops
+    # Sets up Slack object that will send notifications to Slack via a webhook.
+    class Slack
+      def initialize(channel, url, user, message, options = {})
+        @channel = channel
+        @url = url
+        @user = user
+        @message = message
+        @options = options
+      end
+
+      def notify
+        notifier = ::Slack::Notifier.new @url, http_options: @options[:http_options]
+
+        target = if @message[:branch]
+                   @message[:branch]
+                 elsif @message[:module]
+                   @message[:module]
+                 end
+
+        msg = format_message(target)
+
+        notifier.post text: msg[:fallback],
+                      channel: @channel,
+                      username: @user,
+                      icon_emoji: @options[:icon_emoji],
+                      attachments: [msg]
+      end
+
+      private
+
+      def format_message(target)
+        message = {
+          author: 'r10k for Puppet',
+          title: "r10k deployment of Puppet environment #{target}"
+        }
+
+        case @message[:status_code]
+        when 202
+          message.merge!(
+            color: 'good',
+            text: "Successfully deployed #{target}",
```

Comment:
202 is technically just forked and can still fail. Should this be `Deployment started`?